src/averageAges.js

```diff
@@ -55,7 +76,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f'
+    && (people.some(human => human.mother === person.name)));
+
+  const children = people
+    .filter(child => mothers.some(mother => child.mother === mother.name)
+    && (!onlyWithSon || child.sex === 'm'));
+
+  const ageDifferences = children.map(child => {
+    const foundMom = mothers.find(mom => mom.name === child.mother);
+
+    return foundMom ? child.born - foundMom.born : null;
```

Comment:
Return 0 instead of null if the mother is not found to avoid potential issues in further calculations